ExporterManagerTest.shouldRemoveExporterFromState is flaky

```java

[2019-04-25T13:43:16.683Z] [ERROR] shouldRemoveExporterFromState(io.zeebe.broker.exporter.ExporterManagerTest)  Time elapsed: 30.076 s  <<< FAILURE!

[2019-04-25T13:43:16.683Z] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>

[2019-04-25T13:43:16.683Z] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)

[2019-04-25T13:43:16.683Z] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)

[2019-04-25T13:43:16.683Z] 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)

[2019-04-25T13:43:16.683Z] 	at io.zeebe.test.util.TestUtil$Invocation.until(TestUtil.java:96)

[2019-04-25T13:43:16.683Z] 	at io.zeebe.test.util.TestUtil$Invocation.until(TestUtil.java:69)

[2019-04-25T13:43:16.683Z] 	at io.zeebe.test.util.TestUtil.waitUntil(TestUtil.java:41)

[2019-04-25T13:43:16.683Z] 	at io.zeebe.broker.exporter.ExporterManagerTest.shouldRemoveExporterFromState(ExporterManagerTest.java:105)

[2019-04-25T13:43:16.683Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

[2019-04-25T13:43:16.683Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

[2019-04-25T13:43:16.683Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

[2019-04-25T13:43:16.683Z] 	at java.lang.reflect.Method.invoke(Method.java:498)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)

[2019-04-25T13:43:16.683Z] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)

[2019-04-25T13:43:16.683Z] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)

[2019-04-25T13:43:16.683Z] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)

[2019-04-25T13:43:16.683Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)

[2019-04-25T13:43:16.683Z] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)

[2019-04-25T13:43:16.683Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)

[2019-04-25T13:43:16.683Z] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.Suite.runChild(Suite.java:128)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.Suite.runChild(Suite.java:27)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

[2019-04-25T13:43:16.683Z] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)

[2019-04-25T13:43:16.683Z] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

```

--- a/broker-core/src/main/java/io/zeebe/broker/exporter/ExporterManagerService.java
+++ b/broker-core/src/main/java/io/zeebe/broker/exporter/ExporterManagerService.java
@@ -127,8 +127,13 @@ public class ExporterManagerService implements Service<ExporterManagerService> {
       snapshotController.takeSnapshot(snapshotPosition + 1);
 
     } catch (Exception e) {
-      e.printStackTrace();
       LOG.error("Failed to remove exporters from state", e);
+    } finally {
+      try {
+        snapshotController.close();
+      } catch (Exception e) {
+        LOG.error("Unexpected exception happened on closing snapshot controller.", e);
+      }
     }
   }
 

--- a/logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java
+++ b/logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java
@@ -270,6 +270,11 @@ public class StreamProcessorController extends Actor {
     }
   }
 
+  @Override
+  protected void onActorClosed() {
+    LOG.debug("Closed stream processor controller {}.", getName());
+  }
+
   private void onFailure() {
     phase = Phase.FAILED;
 

--- a/logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java
+++ b/logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java
@@ -241,7 +241,9 @@ public class StateSnapshotController implements SnapshotController {
   @Override
   public ZeebeDb openDb() {
     if (db == null) {
-      db = zeebeDbFactory.createDb(storage.getRuntimeDirectory());
+      final File runtimeDirectory = storage.getRuntimeDirectory();
+      db = zeebeDbFactory.createDb(runtimeDirectory);
+      LOG.debug("Opened database from '{}'.", runtimeDirectory.toPath());
     }
 
     return db;
@@ -275,6 +277,8 @@ public class StateSnapshotController implements SnapshotController {
   public void close() throws Exception {
     if (db != null) {
       db.close();
+      final File runtimeDirectory = storage.getRuntimeDirectory();
+      LOG.debug("Closed database from '{}'.", runtimeDirectory.toPath());
       db = null;
     }
   }

